### PR TITLE
merge QA upgrade scripts update to 1.15

### PR DESCRIPTION
**Pull request by lilachmaliniak**
Enhanced script to enable downgrade and upgradeLive script update

### CODE REVIEW

File: deployment/upgradeLive [hunk 1]
```diff
@@ -3,15 +3,32 @@
 # This uploads NVM
 set -e
 [ -s "$NVM_DIR/nvm.sh" ] && . "$NVM_DIR/nvm.sh" || (echo "nvm not found in $NVM_DIR, this is a must, Exiting!" ; exit 1 )
+FORCE=
+REPLACE_SCRIPT=
 cd /opt/kaltura/liveController
+if [ $# -eq 2 ]; then
+ if [ "$2" = "force" ]; then
```
Comment (gadyaari) [inline deployment/upgradeLive]: I think this won't work. You're saying "if number of parameters is NOT 2 --> parameter num 2 = something"

File: deployment/upgradeLive [hunk 2]
```diff
@@ -22,23 +48,81 @@ if [ ! -d "$1" ] ; then
...
+            replace_script)
+                update_upgradeLive_script $1
+                exit 0
+                ;;
+            downgrade)
+                stop
+                downgrade $1
+                ;;
+                    *)
+                        if [ "$PARAM" != "install" ]; then
```
Comment (gadyaari) [inline deployment/upgradeLive]: remove condition

File: deployment/upgradeLive [hunk 3]
```diff
@@ -22,23 +48,81 @@ if [ ! -d "$1" ] ; then
...
-    /etc/init.d/liveRecorder restart
+    if [ -r /etc/init.d/LiveRecorder ]; then
+          if [ ! -L /etc/init.d/LiveRecorder ]; then
+                echo "Creating symlink to LiveRecorder script at - /etc/init.d/LiveRecorder"
+                ln -s /opt/kaltura/liveController/latest/liveRecorder/serviceWrappers/linux/LiveRecorder /etc/init.d/LiveRecorder
+          fi
+          echo "starting liveRecorder..."
+          /etc/init.d/liveRecorder start
+    else
+        echo "this machin dosn't run liveRecorder"
```
Comment (gadyaari) [inline deployment/upgradeLive]: machine

File: deployment/upgradeLive [hunk 4]
```diff
@@ -22,23 +48,81 @@ if [ ! -d "$1" ] ; then
...
+                        if [ "$PARAM" != "install" ]; then
+                                echo "unknown live upgrade option \"${PARAM}\". Optional flags {replace_script|downgrade}"
+                        fi
+                exit 1
+                ;;
+        esac
+
+        start $1
 else
     echo $1 is already exists
```
Comment (gadyaari) [inline deployment/upgradeLive]: change print to be more informative

File: deployment/upgradeLive [hunk 5]
```diff
@@ -2,43 +2,144 @@
...
-    mv liveDVR-$1 $1
-    rm -rf /opt/kaltura/liveController/v$1.tar.gz
-    cd $1
-    echo "current dir: [`pwd`]"
+if [ $# -eq 2 ]; then
+        PARAM=$2
+fi
+
+function download_new_version() {
+	rm -rf "v$VERSION.tar.gz.1*"
```
Comment (gadyaari) [inline deployment/upgradeLive]: don't understand this line

File: deployment/upgradeLive [hunk 6]
```diff
@@ -2,43 +2,133 @@
...
+        install)
+            echo "upgrading liveController to v${VERSION}"
+            stop
+            download_new_version
+            install
+            start
+            ;;
+        replace_script)
+            echo "replacing upgradeLive script to v${VERSION}"
+            replace_upgrade_script $VERSION
```
Comment (gadyaari) [inline deployment/upgradeLive]: what's the meaning of the variable at the end of the line?
$VERSION is defined above and is know by all functions

File: deployment/upgradeLive [hunk 7]
```diff
@@ -2,43 +2,133 @@
...
+    ln -s /opt/kaltura/liveController/$VERSION /opt/kaltura/liveController/latest
+
+}
+
+function downgrade() {
+    if [ -L "latest" ]; then
+        echo "unlink latest"
+        unlink latest
+    fi
+    ln -s /opt/kaltura/liveController/$VERSION /opt/kaltura/liveController/latest
```
Comment (gadyaari) [inline deployment/upgradeLive]: we can't assume that machine has the desired version folder. Maybe a day before someone cleaned all the folder from previous version.
In that case we will have to download again the required version

**Comment by lilachmaliniak** [comment]
updated QA upgrade script. 
Highlights of new features:
- downgrad
- replace updateLive script

**Pull request merged**